From 2a39ef6fe407592efe223ec8ce177262eac5c47b Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 15 Dec 2021 05:10:48 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: security: set axprot for vdec"

Moving axprot configuration of vdec to zephyr post isolation.
This is required as the engines which have isolation logic must
be accessed/configured post isolation configuration.

This reverts commit bfd1a76b08cf0c09ec348a168002f52749111729.

Change-Id: Iecc3c66617b744e534d4ebab5cdff319cdbc5223
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 143c913..5393328 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -204,7 +204,6 @@
 };
 
 enum preboot_sec_axprot_regs {
-	AXPROT_VDEC,
 	AXPROT_ROT,
 	AXPROT_MME,
 	AXPROT_PDMA,
@@ -918,8 +917,6 @@
 #define AXPROT_OVRD_EN		BIT(0)
 #define AXPROT_USER_VAL		((PROT_OVR_USER << 1) | AXPROT_OVRD_EN)
 #define AXPROT_SEC_VAL		((PROT_OVR_SECURE << 1) | AXPROT_OVRD_EN)
-#define VDEC_RTR_INST_NUM	2
-#define VDEC_RTR_INST_DIST	0x10000
 #define ROT_RTR_INST_NUM	1
 #define ROT_INST_NUM		2
 #define ROT_INST_DIST		0x10000
@@ -936,14 +933,6 @@
 #define PDMA_INST_DIST		0x10000
 
 static struct axprot_regs axprot_reg[] = {
-	[AXPROT_VDEC] = {
-		.base_reg = mmDCORE0_VDEC0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
-		.value = AXPROT_USER_VAL,
-		.dcore_dist = DCORE_DIST,
-		.dcore_num = NUM_OF_DCORES,
-		.inst_dist = VDEC_RTR_INST_DIST,
-		.inst_num = VDEC_RTR_INST_NUM,
-	},
 	[AXPROT_ROT] = {
 		.base_reg = mmROT0_MSTR_IF_CORE_HBW_AXPROT_OVRD,
 		.value = AXPROT_USER_VAL,
@@ -982,18 +971,6 @@
 			regmap_writel(reg->value, rtr_reg);
 		}
 	}
-}
-
-static void preboot_secure_vdec(void)
-{
-
-	/* HBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_VDEC);
-
-	axprot_reg[AXPROT_VDEC].base_reg = mmDCORE0_VDEC0_MSTR_IF_CORE_LBW_AXPROT_OVRD;
-
-	/* LBW AXPROT */
-	preboot_axprot_regs_init(AXPROT_VDEC);
 }
 
 static void preboot_secure_rot(void)
@@ -1075,7 +1052,6 @@
 	if (preboot_security_is_enabled()) {
 		preboot_lbw_rtr_preconfig();
 		hl_dev_prot_bits_init();
-		preboot_secure_vdec();
 		preboot_secure_rot();
 		preboot_secure_pmmu();
 		preboot_secure_arc_farm();
